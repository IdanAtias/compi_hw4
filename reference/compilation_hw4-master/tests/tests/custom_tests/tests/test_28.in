// Test operation precedence
int a = 2;
int b = 5;

// Addition and subtraction have same precedence so the earlier should operate first
a - b + a;
a + b - a;

// Multiplication and division go before addition and subtraction
a - a*b;
b + b/b;
a * b / a;
a / a * b;

// Unary minus comes before arithmetic operations
a + -b;
a + --b;
a + ---b;
b / -a;
b / --a;
b / ---a;

// Power comes before unary minus
// UPDATE: Power is not defined for scalars
//1 + -2^(4);
//a - b + -a^b;


matrix m1(3,3) = [1, 2, 3; 4, 5, 6; 7, 8, 9];
matrix m2(3,3) = [3, 3, 3; 2, 2, 2; 1, 1, 1];
matrix m3(3,2) = [1, 2; 4, 5; 7, 8];
matrix m4(3,1) = [3; 6; 9];

m1 - m2 + m1;
m1 + m2 - m1;

m3::m4 + m2;

m1 - m1*m2;
m1 + m1*m2;

// Concatanation comes before multiplication
m1*m2::m4;
m4::m1*[1, 2; 3, 4; 5, 6; 7, 8];

m1 + -m2;
m1 + --m2;
m1 + ---m2;
m1 * -m2;
m1 * --m2;
m1 * ---m2;

m1 + -m1 ^ (3);

// Tranpose comes before everything else
transpose(m1) ^ (2);

m1 - m2 + m3::m4 + -m2 * transpose(m3::m4) ^ (2);

