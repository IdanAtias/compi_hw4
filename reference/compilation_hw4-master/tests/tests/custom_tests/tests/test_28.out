a=
2
b=
5
-1
5
-8
6
5
5
-3
7
-3
-2
2
-2
m1=
1,2,3
4,5,6
7,8,9
m2=
3,3,3
2,2,2
1,1,1
m3=
1,2
4,5
7,8
m4=
3
6
9
-1,1,3
6,8,10
13,15,17
3,3,3
2,2,2
1,1,1
4,5,6
6,7,8
8,9,10
-9,-8,-7
-24,-23,-22
-39,-38,-37
11,12,13
32,33,34
53,54,55
10,10,10,42
28,28,28,96
46,46,46,150
37,46
85,106
133,166
-2,-1,0
2,3,4
6,7,8
4,5,6
6,7,8
8,9,10
-2,-1,0
2,3,4
6,7,8
-10,-10,-10
-28,-28,-28
-46,-46,-46
10,10,10
28,28,28
46,46,46
-10,-10,-10
-28,-28,-28
-46,-46,-46
-467,-574,-681
-1058,-1300,-1542
-1649,-2026,-2403
30,66,102
36,81,126
42,96,150
-325,-728,-1131
-210,-478,-746
-95,-228,-361
